### PR TITLE
follow accepted CRD API best-practices, generate clients

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/area prow
/kind feature
/kind tech-debt
/cc @BenTheElder @krzyzacy @kargakis 
/assign @cjwagner

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
> k8s-ci-robot added size/XXL labels 2 minutes ago

/meow space

**Comment by stevekuznetsov** [comment]
@krzyzacy these clients will allow us to write controllers on `ProwJob`s with ease, which unblocks a status controller to report to GitHub, etc.

**Comment by krzyzacy** [comment]
yay! thanks!

File: prow/apis/prowjobs/v1alpha1/register.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"k8s.io/test-infra/prow/apis/prowjobs"
+)
+
+// SchemeGroupVersion is group version used to register these objects
+var SchemeGroupVersion = schema.GroupVersion{Group: prowjobs.GroupName, Version: "v1alpha1"}
```
[the remark this answers is not in the document]
Comment (kargakis) [reply prow/apis/prowjobs/v1alpha1/register.go]: Btw will the generated clients continue to list v1 prowjobs?

**Comment by kargakis** [comment]
Finally!

File: prow/apis/prowjobs/v1alpha1/register.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kargakis) [reply prow/apis/prowjobs/v1alpha1/register.go]: > Oh eek this should be v1

OK cool

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply prow/apis/prowjobs/v1alpha1/register.go]: Yeah we need to start at v1 and move forward later

**Comment by stevekuznetsov** [comment]
Could not get this to work:`bazel test //hack:verify-codecgen` target... using `$TEST_SRCDIR`, `$TEST_WORKSPACE` and depending on `//:WORKSPACE` all were not possible to find the root of the repo ... so I made a different presubmit.

**Comment by spiffxp** [comment]
fine, space cat it is
/meow space

**Comment by stevekuznetsov** [comment]
@cjwagner @fejta PTAL

**Comment by stevekuznetsov** [comment]
Seems like there was general consensus we could go forward with this today on Slack. @cjwagner @fejta anything else needed here? I will follow-up with client updates

**Comment by krzyzacy** [comment]
@stevekuznetsov one thing worth mention is that, @pwittrock mentioned yesterday that we now have dynamic k8s client available now, which won't involve codegen anymore. Maybe wanna also take a look at that? (@pwittrock can you attach a link?)

**Comment by fejta** [comment]
I'm good with this btw

**Comment by stevekuznetsov** [comment]
@fejta thanks!
@cjwagner you hold the LGTM keys to this castle then!

**Comment by kargakis** [comment]
finally! :tada:

**Pull request merged**